### PR TITLE
Created dense_to_sparse in contrib.layers

### PR DESCRIPTION
**Pull request by jvgonzalvo**
Added `dense_to_sparse`. This does the conversion of dense labels into sparse ones to be passed into the core ctc_loss function. Addresses feature request https://github.com/tensorflow/tensorflow/issues/15985

### CODE REVIEW

**Review by alextp — CHANGES_REQUESTED** [review]
Isn't it better to have a function to convert ctc labels from dense to sparse?

Also, for all losses the labels argument should come before the inputs argument.

**Comment by selcouthlyBlue** [comment]
By a function converting ctc labels from dense to sparse, do you mean a function just for converting dense tensors to sparse ones?

**Review by jvgonzalvo — COMMENTED** [review]
By a function converting ctc labels from dense to sparse, do you mean a function just for converting dense tensors to sparse ones? I already changed the order of the arguments having labels first before inputs.

**Comment by alextp** [comment]
@selcouthlyBlue yes. I meant that instead of having two loss functions, one which converts some arguments and calls the other,  we could have a simple utility function to convert the argument and a single loss function. This feels to me like a smaller API surface.

**Review by alextp — CHANGES_REQUESTED** [review]
Thanks! Can you add a test now?

**Review by jvgonzalvo — COMMENTED** [review]
I have already removed ctc_loss_dense_labels and replaced it with a utility function dense_to_sparse.

**Review by jvgonzalvo — COMMENTED** [review]
I'm going to need help creating the test cases. This is pretty much my first time contributing here.

**Review by alextp — APPROVED** [review]

**Comment by rmlarsen** [comment]
@selcouthlyBlue Please resolve conflict.

**Comment by selcouthlyBlue** [comment]
I forgot that I was pulling from my copy of the repository. My bad. Fixing the conflicts now.

**Comment by selcouthlyBlue** [comment]
Resolved the conflicts. Test please.

**Comment by rmlarsen** [comment]
@selcouthlyBlue Thanks!

**Comment by selcouthlyBlue** [comment]
It's good to go
